src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -165,10 +186,25 @@ public BAMIndex getIndex() {
         if (mIndex == null) {
             final SAMSequenceDictionary dictionary = getFileHeader()
                     .getSequenceDictionary();
-            mIndex = mEnableIndexCaching ? new CachingBAMFileIndex(mIndexFile,
-                    dictionary, mEnableIndexMemoryMapping)
-                    : new DiskBasedBAMFileIndex(mIndexFile, dictionary,
-                    mEnableIndexMemoryMapping);
+            if (mIndexFile.getName().endsWith(BAMIndex.BAMIndexSuffix)) {
+                mIndex = mEnableIndexCaching ? new CachingBAMFileIndex(mIndexFile,
+                        dictionary, mEnableIndexMemoryMapping)
+                        : new DiskBasedBAMFileIndex(mIndexFile, dictionary,
+                        mEnableIndexMemoryMapping);
+                return mIndex;
+            }
+
+            if (!mIndexFile.getName().endsWith(".crai")) return null;
```

Comment:
We missed this one. Should use the const `CRAI_INDEX_SUFFIX` instead of the string literal.

After that, unless @droazen pops up and comments on `.cram.crai`-write support, or fixing the name of `BAMIndex.BAMIndexSuffix`, we'll just leave this PR as is and merge.